### DISCUSSION TITLE
New Device revolt_zx7717 Power Meter

### DISCUSSION THREAD

**Discussion by ghost** [merbanan/rtl_433#3122] [Ideas]
i found a new device
Revolt ZX-7717 Power Meter

and made a decoder for this
thx Christian for help..
![revolt_zx7717](https://github.com/user-attachments/assets/2807c368-c3eb-43db-9f95-80a8a502cffe)

![revolt_zx7717_output](https://github.com/user-attachments/assets/2975707e-b439-4670-88a8-40be497b83a5)

[revolt-zx7717-code.pdf](https://github.com/user-attachments/files/18253771/revolt-zx7717-code.pdf)
who else has this device and know more about the 2 unknown?
Boing
All Pictures from Pearl.de
https://www.pearl.de/a-ZX7716-3102.shtml
ASIN: B0DQPSNLWL

**Comment by ghost** [comment]
zx7717 has no channel !
a yet unhandled beacon is send at start and then all 5 minutes (+-10sek)
len: 120 (14 bytes)
```
time      : 2025-01-02 11:11:28
model     : Revolt-ZX7717 Device ID : d507
Channel   : 6            Unknown 1 : 5aa0          Unknown 2 : 065c          Current   : 0.338 A
Voltage   : 227.7 V       Power     : 44.5 W        Integrity : CHECKSUM
mpulse_slicer_manchester_zerobit Revolt ZX-7717 power meter  codes: {121}155855f02d02b01d254053885ec0610
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
mrevolt_zx7717_decode unhandled message  codes: {120}0ed5075aa007070000370107000031
```

**Reply by zuckschwerdt** [reply]
`155855f02d02b01d254053885ec0610` shifted is `2ab0abe05a05603a4a80a710bd80c200`which decoded is `0ed5075aa007070000370107000031` -- the beacon has the same pattern but we don't know what the field values are.

**Reply by ghost** [reply]
a second device is on the way.
then we know more.
thx Christian.

for me its good to know: there is NO Channel !
zx7716 Monitor:
Select "Station", then "SET, now start zx7717 Meter and within 5 Seconds its the  new Station (1-6).
unknown_2 hast another meaning.
we will see.

**Comment by ghost** [comment]
One of the yet unknown codes looks very good
11 d507 5aa9 00 64 1d01 065c af03 ea08 9b05 18
11      : length (17)
d507 : ID
5aa9 : unknown_1
0064 : unknown_2  (mb PF)
1d01 : 285 (new unknown_3
065c : unknown_2
af03  : current_A ( 0.943)
ea08 : voltage_V (228.2)
9b05 : power_W (143.5)
18     : CHS ok
This is my cooler P:185W after 15 mins

**Comment by ghost** [comment]
//a small change to logg only  the beacons
	
	if (msg_len != 14) {
        decoder_log_bitrow(decoder, 1, __func__, b, len, "unhandled message");
        return DECODE_FAIL_OTHER; // unhandled message
        }

Meter Coldstart
time : 2025-01-02 21:44:58  Beacon : 0e d507 5aa0 0734 0000 6e02 33 0000 c2 
time : 2025-01-02 21:50:08  Beacon : 0e d507 5aa0 0734 0000 3701 34 0000 8b
time : 2025-01-02 21:55:20  Beacon : 0e d507 5aa0 0735 0000 6e02 34 0000 c4
time : 2025-01-02 22:00:31  Beacon : 0e d507 5aa0 0735 0000 3701 35 0000 8d 
time : 2025-01-02 22:05:43  Beacon : 0e d507 5aa0 0736 0000 6e02 35 0000 c6
time : 2025-01-02 22:10:54  Beacon : 0e d507 5aa0 0736 0000 3701 36 0000 8f
time : 2025-01-02 22:16:06  Beacon : 0e d507 5aa0 0737 0000 6e02 36 0000 c8 
time : 2025-01-02 22:21:18  Beacon : 0e d507 5aa0 0737 0000 3701 37 0000 91
time : 2025-01-02 22:26:29  Beacon : 0e d507 5aa0 0738 0000 6e02 37 0000 ca
time : 2025-01-02 22:31:40  Beacon : 0e d507 5aa0 0738 0000 3701 38 0000 93
time : 2025-01-02 22:36:52  Beacon : 0e d507 5aa0 0739 0000 6e02 38 0000 cc
time : 2025-01-02 22:42:03  Beacon : 0e d507 5aa0 0739 0000 3701 39 0000 95
time : 2025-01-02 22:47:15  Beacon : 0e d507 5aa0 073a 0000 6e02 39 0000 ce 
time : 2025-01-02 22:52:26  Beacon : 0e d507 5aa0 073b 0000 3701 3a 0000 98

not very usefull, but there is a beacon counter which increments +1 every 10 mins
the Monitor 7716 has Energy Management, so mb. the counter is used for this.

**Reply by zuckschwerdt** [reply]
And the counter is in two places. With the known little-endian encoding this could be:
` time : 2025-01-02 22:52:26 Beacon : 0e d507 5aa0 07 3b00 00 3701 3a00 00 98`

If 0x0034 = 52 (= 260 mins) is a cold start, then maybe did you run this sensor for 260 minutes so far? I guess you had it running for days. Will be interesting to see where the new meter starts!

**Reply by ghost** [reply]
The meter takes only 0.6 Watt total so its running day and night. 260mins- sure but some coldstarts
In between to get usable beacon data.
the new sensor gets his own Phase to control L1 and L3. L2 is Not findable..
Now its easy to find other msg lengths, i simply Change 0d>0e>11>12 and 1e
we will see.

**Comment by ghost** [comment]
i have no clue where to find an entry for "50Hz", 0x32 or similar with the given messages..
any ideas ?

**Reply by zuckschwerdt** [reply]
There might be more digits, 50.0 or 50.00 which would be around `f401` or `8813`. Or it could just be a flag for 50/60.

**Comment by ghost** [comment]
There are some cheap solar Stations for car use which, Not connected to an extern net, gives 45, 47,50, even 55 Hz.
This might be a Chance to get this Unknown flag-value or whatever..
edit: i Just viewed all my logs for 0x13, found some but Not plausible..
We will see.

**Reply by ghost** [reply]
the new meter with the ID 0x7e05 (Dezimal 1406)
has a much lower radiaton power.
also the beacon is sendet all 10mins
mb. older software release..
![Revolt-zx7717-2](https://github.com/user-attachments/assets/3d0d1237-3e7e-4ac1-aabc-c4a06206791a)

**Comment by ghost** [comment]
Revolt zx7717 sends on 433Mhz !! ( +-100k)
so, if u hop to 866Mhz with- s1024k , u have a problem. No solution yet..

i made a working script for this problem:
hop_433.sh
#!/bin/bash
x=1
while (x=1)
do
rtl_433 -C si -M level -f 433M -T 62 -F syslog:192.168.2.255:1234
sleep 1
rtl_433 -C si -M level -f 868.95M -s 1024k -T 30 -F syslog:192.168.2.255:1234
sleep 1
done

in future, the -sxxxxk parameter should be separated for each -fxxxxxM
have fun.

**Comment by ghost** [comment]
My final revolt_zx7717 with docs:
zx7717.c.zip
[revolt_zx7717.c.zip](https://github.com/user-attachments/files/18551656/revolt_zx7717.c.zip)

have fun with this cool gadget..

![Revolt-Final](https://github.com/user-attachments/assets/cd58296a-04b1-4541-b378-433613efd3e0)

**Reply by zuckschwerdt** [reply]
Congratulations on the very through analysis! I'll merge this asap.

**Comment by ghost** [comment]
my Debian App for zx7717:
thx Christian, was a nice challenge..
#3122 can be closed.
![energy-1406](https://github.com/user-attachments/assets/9ea0855a-c4f1-4c37-823d-b069b7b7dc77)
![energy-2005](https://github.com/user-attachments/assets/3e92278c-10bf-427f-a53a-a20de47f45f5)
have fun..
